app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,55 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign_to: nil, cat: nil, objects: nil)
+    @assign_to = assign_to
+    @cat = cat
+    @objects = objects
+    @cat_tree = @assign_to.ends_with?("-tags")
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    node[:title] = (object.name.presence || object.description) unless object.kind_of?(MiddlewareServer)
+    node[:hideCheckbox] = false
+    node[:select] = @objects.include?(object.id)
+  end
+
+  def tree_init_options(_tree_name)
+    {
+      :expand => true
+    }
+  end
+
+  def set_locals_for_render
+    locals = super
+    locals.merge!(
+      :oncheck     => "miqOnCheckHandler",
+      :check_url   => "/miq_policy/alert_profile_assign_changed/",
+      :checkboxes  => true,
+      :cfmeNoClick => true,
+      :onclick     => false
+    )
+  end
+
+  def root_options
+    t = @cat_tree ? _("Tags") : ui_lookup(:tables => @assign_to)
+    {
+      :title        => t,
```

Comment:
The `t` is used on a single place, useless allocation. Use directly its value in `:title`.